src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,53 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * 
+ * Represents a Person's address in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ *
+ */
+
+public class Contact {
```

Comment:
Great! You have a super class here. Then you need to add `extends Contact` for subclasses (e.g `Address` `Block` etc. )